internal/data/testdata/trace.go

```diff
@@ -37,28 +32,118 @@ var (
 	TestEndTimestamp = data.TimestampUnixNano(TestEndTime.UnixNano())
 )
 
-func GenerateTraceDataOneEmptyResourceSpans() data.TraceData {
+// TraceTestCase represents one test case with the coresponding TraceData and OTLP representation.
+type TraceTestCase struct {
+	Name      string
+	TraceData data.TraceData
+	OtlpData  []*otlptrace.ResourceSpans
+}
+
+var (
+	NoResourceSpansTraceTestCate = TraceTestCase{
```

Comment:
The same nit repeate below, s/Cate/Case
```suggestion
	NoResourceSpansTraceTestCase = TraceTestCase{
```

Reply:
done.